### ISSUE TITLE
Select template correspondence

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#106]
* As an analyst
* I want to select from letter templates
* so that I can attach them to my correspondence email

**Assumptions & Scope**	
What are the assumptions for this story?
This does not include 

What is in and not in scope?

**Acceptance Criteria**	

**Scenario 1: Open Select Templates**
* GIVEN an analyst is on the Contact Applicant view:
https://jacklyn808742.invisionapp.com/console/share/XU2L77IGEA/648918816
* WHEN the analyst clicks the 'Select Template' button
* THEN a modal will open with a range of templates:
https://jacklyn808742.invisionapp.com/console/share/XU2L77IGEA/648918816

**Scenario 2: Close Select Templates**
* GIVEN an analyst is on the Select Template modal:
https://jacklyn808742.invisionapp.com/console/share/XU2L77IGEA/648918816
* WHEN the analyst clicks the 'X' or the 'Return to Correspondence'  
* THEN the analyst will return to the Contact Applicant view:
https://jacklyn808742.invisionapp.com/console/share/XU2L77IGEA/648918816

**Scenario 3: xxxxxx**
...

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
@JHarrietha-AOT, @m-prodan, @arielleandrews, should an analyst be able to remove an attached file from correspondence? Alternatively they could cancel the correspondence and start again but it they can save drafts likely they would prefer to remove an attachment imo. What do you think? cc @mpilchar

**Comment by m-prodan** [comment]
I think that would be helpful @lmullane  - like in a situation where they selected the wrong template, they may want to just be able to remove it and add another one, instead of cancelling and starting over. Is that what you were thinking?

**Comment by arielleandrews** [comment]
Agreed. Mistakes happen, I think in general it's helpful to be able to undo actions. I can see people selecting the wrong template letter and wanting to easily delete/remove it and then select the correct one

**Comment by JHarrietha-AOT** [comment]
Agreed with both @m-prodan @arielleandrews, it is always a good idea to give users a way to undo their actions. It would be more user friendly to allow users to remove some templates/attachments instead of having to start the process over and waiting for the letters to re-generate. I'll add an icon in the UI to remove individual 
attachments.

**Comment by lmullane** [comment]
Agreed. @JHarrietha-AOT, I'm going to add a scenario where the analyst can remove an attachment.

How would you design this? Thinking something like this:

Option 1

GIVEN the analyst is on the Contact Applicant view
AND there is an attached file
WHEN the applicant clicks the trash icon next to the field
THEN the attached file will be removed
AND the analyst will remain on the Contact Applicant view

Easy enough for them to add back a letter imo, but we could also go with something like Option 2 if you prefer?


GIVEN the analyst is on the Contact Applicant view
AND there is an attached file
WHEN the applicant clicks the trash icon next to the field
THEN a modal will open with the message 'Are you sure you want to delete this attachment?" and a 'Confirm' and 'Cancel' button

WHEN the analyst clicks confirm
THEN the attached file will be removed
AND the analyst will remain on the Contact Applicant view

**Comment by JHarrietha-AOT** [comment]
@lmullane I like option 1, its simple and avoids adding unnecessary clicks.
